src/lib/ScannerRunCommand.ts

```diff
@@ -66,26 +114,36 @@ export abstract class ScannerRunCommand extends ScannerCommand {
 	}
 
 	private async validateFlags(): Promise<void> {
-		// First, perform any validation of the command's specific flags.
-		await this.validateCommandFlags();
+		// First, validate the flags specific to the sub-variant.
+		await this.validateVariantFlags();
 
-		// The output flags are common between subclasses. Validate those too.
-		this.validateOutputFlags();
+		// Then, validate the flags that are common to all variants.
+		await this.validateCommonFlags();
 	}
 
 	/**
 	 * Validate whatever flags are specific to a given implementation of {@link ScannerRunCommand}.
 	 * @protected
 	 * @abstract
 	 */
-	protected abstract validateCommandFlags(): Promise<void>
-
-	/**
-	 * Validate the output-related flags, which are common to all implementations of {@link ScannerRunCommand} and share
-	 * the same constraints.
-	 * @private
-	 */
-	private validateOutputFlags(): void {
+	protected abstract validateVariantFlags(): Promise<void>
+
+
+	private async validateCommonFlags(): Promise<void> {
```

Comment:
Renamed this method, since it does more than just validate output flags now. `--projectdir` validations copied from `scanner:run:dfa`.